server/server.go

```diff
@@ -212,35 +213,41 @@ func (s *Server) handleGetMh(w lookupResponseWriter, r *http.Request) {
 		return
 	}
 
-	findResponse, err := c.Find(ctx, mh)
+	resChan := make(chan model.ProviderResult)
+	errChan := make(chan error)
 
-	if err != nil {
-		s.handleError(w, err, log)
-		return
-	}
+	go c.FindAsync(ctx, mh, resChan, errChan)
```

Comment:
It's unfortunate that `FindAsync` is using [JSON](https://github.com/ipni/go-libipni/blob/main/find/client/http/dhash_client.go#L112) instead of ndjson.

Reply:
Finding encrypted value keys from `dhstore` is instant. It's a single index lookup without cascading involved. What takes the most time - is decrypting each key and doing roundtrips for fetching metadata.

Reply:
I will enable ndjson for that find too in the next PR